test/UnitTests/GitTest.cpp

```diff
@@ -29,36 +98,60 @@ TEST(CGit, RunSet)
 	ASSERT_TRUE(output.Find(_T("windir"))); // should be there on any MS OS ;)
 }
 
-TEST(CGit, IsInitRepos_Git)
+// For performance reason, turn LIBGIT off by default, 
+INSTANTIATE_TEST_CASE_P(CGit, CBasicGitFixture, testing::Values(GIT_CLI, /*LIBGIT,*/ LIBGIT2, LIBGIT2_ALL));
+INSTANTIATE_TEST_CASE_P(CGit, CBasicGitWithEmptyRepositoryFixture, testing::Values(GIT_CLI, /*LIBGIT,*/ LIBGIT2, LIBGIT2_ALL));
+
+TEST_P(CBasicGitFixture, IsInitRepos)
 {
-	CAutoTempDir tmpDir;
+	EXPECT_TRUE(m_Git.IsInitRepos());
+}
 
-	CGit cgit;
-	cgit.m_IsUseLibGit2 = false;
-	cgit.m_CurrentDir = tmpDir.GetTempDir();
+TEST_P(CBasicGitWithEmptyRepositoryFixture, IsInitRepos)
+{
+	CString output;
+	CString testFile = m_Dir.GetTempDir() + L"\\test.txt";
 
-	ASSERT_TRUE(cgit.IsInitRepos());
+	EXPECT_TRUE(m_Git.IsInitRepos());
 
-	CString output;
-	EXPECT_EQ(0, cgit.Run(_T("git.exe init"), &output, CP_UTF8));
+	EXPECT_TRUE(CStringUtils::WriteStringToTextFile((LPCTSTR)testFile, L"this is testing file."));
+	EXPECT_EQ(0, m_Git.Run(_T("git.exe add test.txt"), &output, CP_UTF8));
+	EXPECT_TRUE(output.IsEmpty());
+	output.Empty();
+	EXPECT_EQ(0, m_Git.Run(_T("git.exe commit -m \"Add test.txt\""), &output, CP_UTF8));

```

Comment:
This only works if user.email and user.name are set.
We have two options: Either set it in CBasicGitWithEmptyRepositoryFixture  or we pass it using environment variables in AppVeyor